client/pkg/jointoken/jointoken.go

```diff
@@ -0,0 +1,54 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+// Package jointoken implements siderolink jointoken parser.
+package jointoken
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"strings"
+)
+
+// EncodedPrefix is used for the tokens encoded in base64.
+const EncodedPrefix = "b64:"
+
+// JoinToken is the siderolink join token.
+// Custom type adds methods for encoding/decoding extra data from the token.
+type JoinToken struct {
+	ExtraData map[string]string `json:"extra_data"`
```

Comment:
I wonder if we should instead of just passing extra data completely unverified, use HMAC with the password to simply "sign" the data? This way the join token can't modified without knowing the password, and it is still verifiable/

Reply:
yep, makes sense